server/src/main/java/org/elasticsearch/cluster/coordination/ElasticsearchNodeCommand.java

```diff
@@ -177,6 +177,17 @@ protected void cleanUpOldMetaData(Terminal terminal, Path[] dataPaths, long newG
         MetaData.FORMAT.cleanupOldFiles(newGeneration, dataPaths);
     }
 
+    protected NodeEnvironment.NodePath[] toNodePaths(Path[] dataPaths) {
+        return Arrays.stream(dataPaths).map(ElasticsearchNodeCommand::createNodePath).toArray(NodeEnvironment.NodePath[]::new);
+    }
+
+    private static NodeEnvironment.NodePath createNodePath(Path path) {
+        try {
+            return new NodeEnvironment.NodePath(path);
+        } catch (IOException e) {
+            throw new ElasticsearchException("Unable to investigate path: " + path + ": " + e.getMessage());
```

Comment:
why not pass `e` as cause to the `ElasticsearchException`?

Reply:
I blindly moved this method here from `NodeRepurposeCommand`, but yes an inner exception is preferable, see e12ca621bed6c02af872bbcbb761756e8630d918.